### ISSUE TITLE
Account specific overrides for user and room names

### ISSUE THREAD

**Issue by deepbluev7** [Nheko-Reborn/nheko#255]
**Is your feature request related to a problem? Please describe.**
Currently assigning a name to a direct chat, sets that name for both participants. It would be useful, if you could name chats (and other users) with a name, that only you can see.

**Describe the solution you'd like**
Save overrides for room and display names in room or global account data, i.e. `im.nheko.custom_name.<room_id>` or `im.nheko.custom_name.<user_id>`. This should be upstreamed to the spec if possible in the future. It may also be extended to room and user avatars, maybe even more data.

**Describe alternatives you've considered**
You could store it in the local config or as a room state event. The former would not sync across client, the latter would be visible for everyone. I think account_data is the right location for it.

**Additional context**
@redsky17 renamed my direct chat room and I want to give it a different name without him noticing.

**Comment by kamathmanu** [comment]
@deepbluev7 Is this still an outstanding issue? I can give it a try if it hasn't been worked on yet.

**Comment by deepbluev7** [comment]
Yes, but it is not that simple, since it needs custom account data objects in mtxclient.

**Comment by Zocker1999NET** [comment]
Hey, I wrote two MSCs which may add this feature to the spec.

- [MSC3865](https://github.com/matrix-org/matrix-spec-proposals/pull/3865), which in general defines user-given attributes for users (including displayname)
- [MSC3864](https://github.com/matrix-org/matrix-spec-proposals/pull/3864), which does the same for room attributes (like room name)

They may both still change in the next weeks, but I expect them to be "stable enough for implementation" in the next month. You may want to join the discussion as you already thought about implementing this feature in nheko.
